File: py/prob_1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num_map = {num:i for i, num in enumerate(nums)}
        res = []
        
        for i in range(len(nums)):
            result = target - nums[i]
            if result in num_map and num_map[result] != i:
                res.extend([i, num_map[result]])
                
                return res    
        return res

File: js/prob_1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
    let num_obj = {}
    for (let i=0; i < nums.length; i++) {
        num_obj[nums[i]] = i
    }
    
    let res = []
    for (let i=0; i < nums.length; i++) {
        let result = target - nums[i]
        
        if (result in num_obj && num_obj[result] != i) {
            res.push(i, num_obj[result])
            
            return res
        }
    }
    return res
}
